src/opening.rs

```diff
@@ -91,6 +94,80 @@ where
 
         self.root == item
     }
+
+    /// Serialize an [`opening`] to a vector of bytes.
+    // Once the new implementation of the `Serializable` trait becomes
+    // available, we will want that instead, but for the time being we use
+    // this implementation.
+    pub fn to_var_bytes<const T_SIZE: usize>(&self) -> Vec<u8>
+    where
+        T: Serializable<T_SIZE>,
+    {
+        let mut bytes = Vec::new();
```

Comment:
Could we at least pre-allocate the number of bytes required to minimize the number of allocations? It seems to me it is going to be:

```text
root:      T_SIZE 
branch:    T_SIZE * H * A
positions: u32::SIZE * H

opening:   T_SIZE + (T_SIZE * A + u32::SIZE) * H
```

```suggestion
        let mut bytes = Vec::with_capacity(T_SIZE + (T_SIZE * A + u32::SIZE) * H);
```